agent/containermetadata/parse_metadata.go

```diff
@@ -0,0 +1,151 @@
+	}
+
+	// Get Port bindings from docker configurations
+	var ports []api.PortBinding
+	var err error
+	// In most cases a container should never lack a host config but we check just in case to avoid
+	// nil pointer exception (Could occur if there is some error in the docker api call)
+	if container.HostConfig != nil {
+		ports, err = api.PortBindingFromDockerPortBinding(container.HostConfig.PortBindings)
+		if err != nil {
+			seelog.Warnf("Failed to parse container metadata: %v", err)
+		}
+	} else {
+		seelog.Warn("Failed to parse container metadata: container has no host configuration")
+	}
+
+	networkMD, err := parseNetworkMetadata(container.NetworkSettings)
+	if err != nil {
+		seelog.Warnf("Failed to parse container metadata: %v", err)
+	}
+
+	dockerContainerMD := DockerContainerMD{
+		containerID:         container.ID,
+		dockerContainerName: container.Name,
+		imageID:             container.Image,
+		imageName:           imageNameFromConfig,
+		ports:               ports,
+		networkInfo:         networkMD,
+	}
+	return dockerContainerMD
+}
+
+// parseTaskMetadata parses  in the AWS configuration and task
+// and packages this data for JSON marshaling
+// Since we accept incomplete metadata fields, we should not return
+// errors here and handle them at this stage.
+func parseTaskMetadata(task *api.Task, container *api.Container) TaskMetadata {
+	containerName := container.Name
+
+	taskARNFromConfig := task.Arn
+
+	return TaskMetadata{
+		containerName: containerName,
+		taskARN:       taskARNFromConfig,
+	}
+}
+
+// parseMetadata gathers metadata from a docker container, and task
+// configuration and data then packages it for JSON Marshaling
+// Since we accept incomplete metadata fields, we should not return
+// errors here and handle them at this stage.
+func (manager *metadataManager) parseMetadata(createTime time.Time, updateTime time.Time, dockerContainer *docker.Container, task *api.Task, container *api.Container) Metadata {
+	taskMD := parseTaskMetadata(task, container)
+	dockerMD := parseDockerContainerMetadata(dockerContainer)
+	return Metadata{
+		cluster:                 manager.cluster,
+		taskMetadata:            taskMD,
+		dockerContainerMetadata: dockerMD,
+		containerInstanceARN:    manager.containerInstanceARN,
+		metadataStatus:          "Updated",
+	}
+}
+
+// parseMetadataAtContainerCreate gathers metadata from task and cluster configurations
+// then packages it for JSON Marshaling. We use this version to get data
+// available prior to container creation
+// Since we accept incomplete metadata fields, we should not return
+// errors here and handle them at this stage.
+func (manager *metadataManager) parseMetadataAtContainerCreate(task *api.Task, container *api.Container) Metadata {
+	taskMD := parseTaskMetadata(task, container)
+	return Metadata{
+		cluster:              manager.cluster,
+		taskMetadata:         taskMD,
+		containerInstanceARN: manager.containerInstanceARN,
+		metadataStatus:       "Created",
```

Comment:
"NOT_READY" and "READY" seem to be better names since the "Created" state means the container has not started or our update failed for whatever reason.